projects/opendr_ws_2/src/opendr_perception/README.md

```diff
@@ -0,0 +1,874 @@
+      - `--backbone BACKBONE`: Backbone network (default=`darknet53`)
+
+   3. YOLOv5 node
+      ```shell
+      ros2 run opendr_perception object_detection_2d_yolov5_node.py
+      ```
+      The following optional argument is available for the YOLOv5 node:
+      - `--model_name MODEL_NAME`: Network architecture, options are `yolov5s`, `yolov5n`, `yolov5m`, `yolov5l`, `yolov5x`, `yolov5n6`, `yolov5s6`, `yolov5m6`, `yolov5l6`, `custom` (default=`yolov5s`)
+
+   4. CenterNet node
+      ```shell
+      ros2 run opendr_perception object_detection_2d_centernet_node.py
+      ```
+      The following optional argument is available for the CenterNet node:
+      - `--backbone BACKBONE`: Backbone network (default=`resnet50_v1b`)
+
+   5. Nanodet node
+      ```shell
+      ros2 run opendr_perception object_detection_2d_nanodet_node.py
+      ```
+      The following optional argument is available for the Nanodet node:
+      - `--model Model`: Model that config file will be used (default=`plus_m_1.5x_416`)
+
+   6. DETR node
+      ```shell
+      ros2 run opendr_perception object_detection_2d_detr_node.py
+      ```
+
+   The following optional arguments are available for all nodes above:
+   - `-h or --help`: show a help message and exit
+   - `-i or --input_rgb_image_topic INPUT_RGB_IMAGE_TOPIC`: topic name for input RGB image (default=`/image_raw`)
+   - `-o or --output_rgb_image_topic OUTPUT_RGB_IMAGE_TOPIC`: topic name for output annotated RGB image, `None` to stop the node from publishing on this topic (default=`/opendr/image_objects_annotated`)
+   - `-d or --detections_topic DETECTIONS_TOPIC`: topic name for detection messages, `None` to stop the node from publishing on this topic (default=`/opendr/objects`)
+   - `--device DEVICE`: Device to use, either `cpu` or `cuda`, falls back to `cpu` if GPU or CUDA is not found (default=`cuda`)
+
+3. Default output topics:
+   - Output images: `/opendr/image_objects_annotated`
+   - Detection messages: `/opendr/objects`
+
+   For viewing the output, refer to the [notes above.](#notes)
+
+### 2D Single Object Tracking ROS2 Node
+
+You can find the single object tracking 2D ROS2 node python script [here](./opendr_perception/object_tracking_2d_siamrpn_node.py) to inspect the code and modify it as you wish to fit your needs.
+The node makes use of the toolkit's [single object tracking 2D SiamRPN tool](../../../../src/opendr/perception/object_tracking_2d/siamrpn/siamrpn_learner.py) whose documentation can be found [here](../../../../docs/reference/object-tracking-2d-siamrpn.md).
+
+#### Instructions for basic usage:
+
+1. Start the node responsible for publishing images. If you have a USB camera, then you can use the `usb_cam_node` as explained in the [prerequisites above](#prerequisites).
+
+2. You are then ready to start the single object tracking 2D node:
+
+    ```shell
+    ros2 run opendr_perception object_tracking_2d_siamrpn_node.py
```

Comment:
Got this error:
```
Model name: yolo_default --> yolo_default/yolo_default.json
[10:04:31] ../src/base.cc:80: cuDNN lib mismatch: linked-against version 8600 != compiled-against version 8100.  Set MXNET_CUDNN_LIB_CHECKING=0 to quiet this warning.
Loaded parameters and metadata.
[INFO] [1671786273.835652375] [opendr_object_tracking_2d_siamrpn_node]: Object Tracking 2D SiamRPN node started.
Traceback (most recent call last):
  File "/home/daniel/opendr/projects/opendr_ws_2/install/opendr_perception/lib/opendr_perception/object_tracking_2d_siamrpn", line 33, in <module>
    sys.exit(load_entry_point('opendr-perception==1.0.0', 'console_scripts', 'object_tracking_2d_siamrpn')())
  File "/home/daniel/opendr/projects/opendr_ws_2/install/opendr_perception/lib/python3.8/site-packages/opendr_perception/object_tracking_2d_siamrpn_node.py", line 169, in main
    rclpy.spin(object_tracker_2d_siamrpn_node)
  File "/opt/ros/foxy/lib/python3.8/site-packages/rclpy/__init__.py", line 191, in spin
    executor.spin_once()
  File "/opt/ros/foxy/lib/python3.8/site-packages/rclpy/executors.py", line 718, in spin_once
    raise handler.exception()
  File "/opt/ros/foxy/lib/python3.8/site-packages/rclpy/task.py", line 239, in __call__
    self._handler.send(None)
  File "/opt/ros/foxy/lib/python3.8/site-packages/rclpy/executors.py", line 429, in handler
    await call_coroutine(entity, arg)
  File "/opt/ros/foxy/lib/python3.8/site-packages/rclpy/executors.py", line 354, in _execute_subscription
    await await_or_execute(sub.callback, msg)
  File "/opt/ros/foxy/lib/python3.8/site-packages/rclpy/executors.py", line 118, in await_or_execute
    return callback(*args)
  File "/home/daniel/opendr/projects/opendr_ws_2/install/opendr_perception/lib/python3.8/site-packages/opendr_perception/object_tracking_2d_siamrpn_node.py", line 108, in callback
    self.tracker.infer(self.image, init_box)
  File "/home/daniel/opendr/src/opendr/perception/object_tracking_2d/siamrpn/siamrpn_learner.py", line 323, in infer
    self.tracker.init(img, gt_bbox_, ctx=self.ctx)
  File "/home/daniel/opendr/venv/lib/python3.8/site-packages/gluoncv/model_zoo/siamrpn/siamrpn_tracker.py", line 449, in init
    z_crop = self.get_subwindow(img, self.center_pos,
  File "/home/daniel/opendr/venv/lib/python3.8/site-packages/gluoncv/model_zoo/siamrpn/siamrpn_tracker.py", line 414, in get_subwindow
    im_patch = cv2.resize(im_patch, (model_sz, model_sz))
cv2.error: OpenCV(4.5.1) /tmp/pip-req-build-ms668fyv/opencv/modules/imgproc/src/resize.cpp:4051: error: (-215:Assertion failed) !ssize.empty() in function 'resize'

```

Reply:
In my local system with a laptop camera it runs properly, it is probably related to the input size, i'll look into it.

Reply:
I asked Vivi as she is the original author and she suggest adding a minimum size for the bounding box as a possible fix, and also added a log for the shape of the image in case something goes wrong with the input. 

@ad-daniel would you be able to check again as i cannot reproduce the issue?

Reply:
seems to have done the trick :+1:

Reply:
Spot on, thanks @vivinousi